src/components/FilterAndSortPopUp/FilterAndSortPopUp.jsx

```diff
@@ -1,50 +1,65 @@
-                    <input type="text" name="sortNotes" defaultValue="Newest First" list="sortList" id="tn_sort" />
-                    <datalist id="sortList">
-                        <option value="Newest First" />
-                    </datalist>
-
-                    <label htmlFor="tn_filter" className="tn_popUp-subheading" >Filter By</label>
-                    <input type="text" name="filterNotes" defaultValue="Labels" list="filterList" id="tn_filter" />
-                    <datalist id="filterList">
-                        <option value="Labels" />
-                        <option value="Pinned" />
-                        <option value="Archived" />
-                    </datalist>
-
-                    <div className="tn_popUp-subheading" >Select Labels</div>
-                    <label className="tn_select-label" htmlFor="label-1">
-                        <input type="checkbox" id="label-1" />
-                        Label 1
-                    </label>
-                    <label className="tn_select-label" htmlFor="label-2">
-                        <input type="checkbox" id="label-2" />
-                        Label 2
-                    </label>
-                    <label className="tn_select-label" htmlFor="label-3">
-                        <input type="checkbox" id="label-3" />
-                        Label 3
-                    </label>
+                    <label htmlFor="tn_sort" className="tn_popUp-subheading" >Sort By Date</label>
+                    <select name="sortNotes" id="tn_sort" value={filterAndSortState.SortByDate} onChange={(e) => filterAndSortDispatch({ type: "SORT_BY_DATE", payload: e.target.value })} >
+                        <option value=""></option>
+                        <option value="Newest First">Newest First</option>
+                        <option value="Oldest First">Oldest First</option>
+                    </select>
+
+                    <label htmlFor="tn_filter" className="tn_popUp-subheading" >Filter By Category</label>
+                    <select name="filterNotes" id="tn_filter" value={filterAndSortState.FilterByCategory} onChange={(e) => filterAndSortDispatch({ type: "FILTER_BY_CATEGORY", payload: e.target.value })} >
+                        <option value=""></option>
+                        <option value="Labels" >Labels</option>
+                        <option value="Pinned" >Pinned</option>
+                        <option value="Archived" >Archived</option>
+                    </select>
+
+                    {
+                        labeledNotes.length > 0 ?
+                            <>
+                                <div className="tn_popUp-subheading" >Select Labels</div>
+                                {
+                                    uniqueLabels.map((label, index) => {
+                                        return (
+                                            <label className="tn_select-label" htmlFor={label} key={index}>
```

Comment:
avoid using index for` key`

Reply:
okk will change it!! and will use .id instead!!